### ISSUE TITLE
Modify Markdown `dl` syntax to support multiple subsequent `dt` or `dd` elements

### ISSUE THREAD

**Issue by develleoper** [naturalcrit/homebrewery#2340]
## Description
The current `<dl>` syntax (i.e. `dt :: dd`) only supports a 1:1 ratio of terms to descriptions.

### Reference
- Original Discussion at [Reddit](https://www.reddit.com/r/homebrewery/comments/tdre2d/feature_request_support_dl_tag_syntax_extension/)
- The following examples follow the syntax set forth in [Kramdown](https://kramdown.gettalong.org/syntax.html#definition-lists), [Pandoc](https://pandoc.org/MANUAL.html#definition-lists), [MultiMarkdown](http://fletcher.github.io/MultiMarkdown-4/definitionlists), and [Markdown Extra](https://michelf.ca/projects/php-markdown/extra/#def-list).

#### Basic Case:
```markdown
First term
: First description
: Second description
```
```html
<dl>
  <dt>First term</dt>
  <dd>First description</dd>
  <dd>Second description</dd>
</dl>
```

**Comment by develleoper** [comment]
#### Advanced Cases:
```markdown
Preceding paragraph

First term
Second term
: Description, first paragraph
  Description, second paragraph
  - Description, unordered list item
  > Description, block quote

Following paragraph
```
```html
<p>Preceding paragraph</p>
<dl>
  <dt>First term</dt>
  <dt>Second Term</dt>
  <dd>
    <p>Description, first paragraph</p>
    <p>Description, second paragraph</p>
    <ul>
      <li>Description, unordered list item</li>
    </ul>
    <blockquote>Description, block quote</blockquote>
  </dd>
</dl>
<p>Following paragraph</p>
```

**Comment by calculuschild** [comment]
What would you see this be displayed as when entered into the Homebrewery? Any particular style/format from the D&D books that a multi-dd would be used for?

**Comment by ericscheid** [comment]
Two paragraphs in a `<dd>`, from #181 

<img width="426" alt="two paragraphs in a dd" src="https://user-images.githubusercontent.com/2605467/188527307-6bb4ca9d-5263-48ca-8668-426f6992af91.png">

**Comment by develleoper** [comment]
@ericscheid I believe the ticket and associated pull request actually use bold tags and paragraphs to accomplish that? Is this incorrect?

@calculuschild honestly, there's nothing here we can't reproduce via bold tags and default paragraph indents _(or just manually writing one's own `<dl>` tags)_; but I do think that the image shared by @ericscheid is one good example of its potential use-case, and the emerged standard used across the other various markdown implementations provides a more succinct and consistent syntax. Personally, in development work, I often use them for elements with one header and multiple inline values, like the lists of skills and equipment which classes choose from at level 1.

**Comment by calculuschild** [comment]
> Two paragraphs in a <dd>, from https://github.com/naturalcrit/homebrewery/issues/181

This is already handled naturally by just using normal paragraphs, since we don't use `dd` in that part of the stat block.

![image](https://user-images.githubusercontent.com/5878534/188742733-9ab46589-dbaa-4e56-a948-fc560623ed9b.png)

@develleoper One issue we had in  the original version of the Homebrewery was over-use of obscure syntax combinations to activate different features (two `<hr>` next to a blockquote would appear as a stat block, code blocks using backticks being invisible and used for column breaks, etc. It got really easy to write bad code that led to lots of user error, and we also very quickly ran out of enough syntax to do everything we wanted. The V3 renderer was an attempt to clear out a lot of "unused" syntax to make room for cleaner stuff. For instance we don't use blockquotes `>` at all in V3 because they were obnoxious to type out and we could get the variety needed just from CSS classes.

So, we're open to adding syntax for stuff, but we do want to make sure it has a clear utility that isn't already covered by other syntax. It's nice to have some syntax (like multi-dd) left as a backup for some future element we haven't accounted for yet. We also don't want multiple syntaxes that do the same thing if we can help it. Once a syntax is out on the site, even if we don't give it any special styling and it just appears as a standard paragraph, we can't really change what it does without breaking existing brews. For example if you are using it just for equipment lists that appear as a standard paragraph anyway, and we change it to appear more as a big blocky stylized box, suddenly all of your brews that relied on the previous appearance will be broken.

**Comment by ericscheid** [comment]
> This is already handled naturally by just using normal paragraphs, since we don't use `dd` in that part of the stat block.

This works because those `dd` in that theme are styled the same as normal paragraphs. This is a workaround.

If however there was another theme that said "definition lists look like this [whatever]" (including multi-paragraph `dd`) then the user would need to come up with a different workaround. Changing themes shouldn't require hacking the brew.markdown just for styling reasons — this is why emitting *semantic* html is useful.

Perhaps @develleoper could work up a PR which merges all three options:

### a simple one-line syntax for one-definition terms:
```
term :: defn
``` 

### multiple definition terms:
```
term::
: defn
: defn2
```

### indented definitions to support complex definitions
```
term::
: defn
  another paragraph
  - bullet item
  - bullet 2
```

For the latter two cases (if possible due to parsing logic) make the `::` optional.

**Comment by calculuschild** [comment]
> This works because those dd in that theme are styled the same as normal paragraphs. This is a workaround.

There are literallly no `dd` present in that part of the statblock. It's not a workaround that styles `dd` to look like a paragraph. We really just use bold text and normal paragraphs:

![image](https://user-images.githubusercontent.com/5878534/188785547-bb0dddfc-441f-4cb2-87b9-d9b80a132d9b.png)

**Comment by ericscheid** [comment]
Yes, sorry, that's what I meant: the "bit of text that is a definition in a list".

Which is a `p` masquerading as  a `dd`.

**Comment by develleoper** [comment]
@calculuschild Yeah, I noticed that the current implementation just uses bold tags and paragraph breaks, and honestly, that _is_ good enough for users to reproduce 5e styles.

I totally understand the battle against syntax bloat, I'm just advocating for altering the _current_ (unused?) `<dl>` support to better comply with the rest of the industry's syntax and to allow for better support of the HTML spec on this tag. The tag is already _in_ the system, it just doesn't support the full feature set that it could, and uses its own unique syntax apart from the one on which other parsers seem to have standardized.

And @ericscheid, I don't want to jump the gun, but will happily work up a PR, if that's acceptable.

**Comment by calculuschild** [comment]
@develleoper 

The current single-line dl syntax is indeed used in multiple places already on the Homebrewery, so we can't eliminate it entirely without breaking existing documents. In fact, it was a deliberate choice to break from the other syntax because the only place where we use DL is a single-dd context, and the `term :: definition` allows the Markdown to be clean-looking in a single line.

Again, we welcome the new syntax (and an accompanying PR), but we would want to have a clear decided use for it. We don't want users to rely on it only to be surprised later on when we finally figure out a use case and start adding styling.

**Comment by ericscheid** [comment]
Yes, it is a term with one definition with two paragraphs — which is part of the proposal.

It would be written as:

````
term
: paragraph one
  paragraph two, on a new line, indented
````

**Comment by develleoper** [comment]
@calculuschild yeah, that totally makes sense. I'm curious why the decision was made to support `<dl>` tags in a one-off way that folks familiar with markdown would not expect, but I respect that y'all have valid reasons, and am mostly just eager to help the parser support both.

I apologize for my initial use-case being so irrelevant to the topic at hand, and do not have

**Comment by develleoper** [comment]
also apologies for the accidentally closed issue, my cat stepped on my touch pad and hit "close with comment" 🤦‍♀️😼

**Comment by ericscheid** [comment]
> I'm curious why the decision was made to support <dl> tags in a one-off way that folks familiar with markdown would not expect

https://github.com/naturalcrit/homebrewery/commit/64d133f8f63cc1a84ec1d65ea184c687b31b3156

The `dl` extension was added as part of the overhaul of transforming the mess of Legacy formatting to the new V3 formatting (we used things like "a bullet list following a horizontal rule will get formatted as X" to do monster stat blocks). 

IIRC, at the time there was no clear markdown-community consensus on how to do definition lists, and Marked didn't have much guidance. We rolled with a simple syntax that worked for the immediate need.

That extension was added in July 2021. The use of that syntax for monster actions was added in Sep 2021 (i.e. the use case in #181 wasn't considered in July)

**Comment by calculuschild** [comment]
That sounds about right. I'll add that I think we may have been "inspired" by the

```
term
: definition
```

syntax somewhere along the way, since we ended up using colons as well. I suspect the reasoning for coming up with our own syntax was something like, "DL syntax isn't in any official specs, and is only a custom extension in a couple of Markdown parsers. Let's tweak it a bit to fit into one line since we will only ever use a single term and definition anyway."

Since Markdown is meant to be ["as readable as possible... publishable as-is, as plain text"](https://spec.commonmark.org/0.30/#what-is-markdown-) , the single-line DL we came up with seemed to match our actual output appearance in a way that remained legible, better than breaking the DL across multiple lines:

```
**Armor Class**
: 20 (chain mail, shield)

**Hit Points**
: 100(1d4 + 5)

**Speed**
: 47ft.
```

vs

```
**Armor Class** :: 20 (chain mail, shield)
**Hit Points**  :: 100(1d4 + 5)
**Speed**       :: 47ft.
```
